### DISCUSSION TITLE
automergeSchedule ignored if platformAutomerge is true

### DISCUSSION THREAD

**Discussion by DnrKtz167** [renovatebot/renovate#21371] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

latest

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Auto-merging should only be done when it is allowed by the automergeSchedule even with platformAutomerge enabled

### What you tried so far.

- Removing platformAutomerge, everything works fine

- Moving platformAutomerge to a package rule from root, did not change the behaviour

### Relevant debug logs

<details><summary>Logs</summary>
DEBUG: Creating PR (repository=my-company/project, branch=renovate/webpack-5.x)
       "title": "fix(deps): update dependency webpack to v5.78.0",
       "head": "my-company:renovate/webpack-5.x",
       "base": "master",
       "draft": false
DEBUG: PR created (repository=my-company/project, branch=renovate/webpack-5.x)
       "pr": 364,
       "draft": false
DEBUG: Adding labels 'dependencies, npm' to #364 (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: GitHub-native automerge: success...PrNo: 364 (repository=my-company/project, branch=renovate/webpack-5.x)
 INFO: PR created (repository=my-company/project, branch=renovate/webpack-5.x)
       "pr": 364,
       "prTitle": "fix(deps): update dependency webpack to v5.78.0"
DEBUG: resolveBranchStatus(branchName=renovate/webpack-5.x, ignoreTests=false) (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: getBranchStatus(renovate/webpack-5.x) (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: branch status check result (repository=my-company/project, branch=renovate/webpack-5.x)
       "state": "pending",
       "statuses": []
DEBUG: Successful checks are all internal renovate/ checks, so returning "pending" branch status (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: No check runs found (repository=my-company/project, branch=renovate/webpack-5.x)
       "result": {"total_count": 0, "check_runs": []}
DEBUG: Branch status yellow (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: Skipping assignees and reviewers as automerge=true (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: setPrCache() (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: Created Pull Request #364 (repository=my-company/project, branch=renovate/webpack-5.x)
DEBUG: PR is configured for automerge (repository=my-company/project, branch=renovate/webpack-5.x)
</details>

**Comment by rarkins** [comment]
Renovate can't control the schedule at which the platform automerges. The only thing which could be done is to clarify this in the docs. PR welcome

**Comment by DnrKtz167** [comment]
Yes, but renovate can obviously enable the platformAutomerge.
In my packageRules I have enabled automerge only for minor and patch versions, for major updates it does not put the PR in the merge queue. However for minor and patch versions it does, so what I'd expect would be the same behaviour if I specify an additional automergeSchedule together with the platformAutomerge, that it only puts the PR into the merge queue when the automergeSchedule allows it.

EDIT: typos & added half a sentence

**Reply by rarkins** [reply]
At the moment Renovate enables platform automerge at PR creation time. Why not just limit PR creation time to the windows when you want automerging?

**Reply by DnrKtz167** [reply]
Since we want to run renovate multiple times a day and still rebase all branches while we work, but use the merge queue in the morning hours to get all PR's merged. We could increase the hourly PR limit and amount of open PR's to counteract that it would only create them in that time frame. 

But I still think that the automergeSchedule should work in addition with the platformAutomerge IMHO.

EDIT: typos

**Reply by rarkins** [reply, marked as answer]
PRs welcome if you would like to have a shot at it

**Reply by DnrKtz167** [reply]
I'll give it a shot after work!

**Reply by rarkins** [reply]
It might be best to start with a feature request issue to discuss, actually. There may be some decisions to make on the logic, such as whether to try _every_ run even if it failed before, etc.
